ui/app/src/components/workspaces/WorkspaceProvider.tsx

```diff
@@ -26,46 +28,75 @@ export const WorkspaceProvider: React.FunctionComponent = () => {
-  const [ apiError, setApiError ] = useState({} as APIError);
+  const [apiError, setApiError] = useState({} as APIError);
   const { workspaceId } = useParams();
 
+  const appRoles = useContext(AppRolesContext);
+  const refIsTREAdminUser = useRef(false);
 
   // set workspace context from url
   useEffect(() => {
     const getWorkspace = async () => {
       try {
         // get the workspace - first we get the scope_id so we can auth against the right aad app
         let scopeId = (await apiCall(`${ApiEndpoint.Workspaces}/${workspaceId}/scopeid`, HttpMethod.Get)).workspaceAuth.scopeId;
-        if (scopeId === "") {
-          console.error("Unable to get scope_id from workspace - authentication not set up.");
-        }
 
-        const ws = (await apiCall(`${ApiEndpoint.Workspaces}/${workspaceId}`, HttpMethod.Get, scopeId)).workspace;
-        workspaceCtx.current.setWorkspace(ws);
-        const ws_application_id_uri = ws.properties.scope_id;
+        const authProvisioned = scopeId !== "";
 
-        // use the client ID to get a token against the workspace (tokenOnly), and set the workspace roles in the context
         let wsRoles: Array<string> = [];
-        console.log('Getting workspace');
-        await apiCall(`${ApiEndpoint.Workspaces}/${workspaceId}`, HttpMethod.Get, ws_application_id_uri, undefined, ResultType.JSON, (roles: Array<string>) => {
-          workspaceCtx.current.setRoles(roles);
-          wsRoles = roles;
-        }, true);
-
-        // get workspace services to pass to nav + ws services page
-        const workspaceServices = await apiCall(`${ApiEndpoint.Workspaces}/${ws.id}/${ApiEndpoint.WorkspaceServices}`, HttpMethod.Get, ws_application_id_uri);
-        setWorkspaceServices(workspaceServices.workspaceServices);
-        setLoadingState(wsRoles && wsRoles.length > 0 ? LoadingState.Ok : LoadingState.AccessDenied);
-
-        // get shared services to pass to nav shared services pages
-        const sharedServices = await apiCall(ApiEndpoint.SharedServices, HttpMethod.Get);
-        setSharedServices(sharedServices.sharedServices);
-
-      } catch (e: any){
-        e.userMessage = 'Error retrieving workspace';
-        setApiError(e);
-        setLoadingState(LoadingState.Error);
+        let ws: Workspace = {} as Workspace;
+
+        if (authProvisioned) {
+          try {
+            ws = (await apiCall(`${ApiEndpoint.Workspaces}/${workspaceId}`, HttpMethod.Get, scopeId)).workspace;
+            workspaceCtx.current.setWorkspace(ws);
+            // use the client ID to get a token against the workspace (tokenOnly), and set the workspace roles in the context
+
+            await apiCall(`${ApiEndpoint.Workspaces}/${workspaceId}`, HttpMethod.Get, ws.properties.scope_id, undefined, ResultType.JSON, (roles: Array<string>) => {
+              workspaceCtx.current.setRoles(roles);
+              wsRoles = roles;
+            }, true);
+          } catch (e: any) {
```

Comment:
This may mask unexpected errors in this block - suggest either not having a wide try statement or catching for the specific error we're expecting if the person doesn't have a role in the workspace scope

Reply:
Ok, there was an unnecessary API call here, didn't realise could just get the roles. Can you take another look.

Reply:
@jjgriff93 can you take a look when you have chance, see if this works better.

Reply:
@jjgriff93 can you give this a sanity check?

Thanks.